### PR TITLE
Upgrade to Apache Beam, version 2.3.0

### PR DESCRIPTION
**Pull request by huygaa11**
Upgrade to Apache Beam, version 2.3.0

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -374,11 +360,6 @@
...
 
     <plugins>
-      <plugin>
-        <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-enforcer-plugin</artifactId>
-      </plugin>
-
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
```
Comment (lukecwik) [inline pom.xml]: beam-examples-java8 is still listed in this file.

File: maven-archetypes/examples/src/test/resources/projects/basic/archetype.properties [hunk 2]
```diff
@@ -16,4 +16,4 @@ package=it.pkg
 version=0.1
 groupId=archetype.it
 artifactId=basic
-targetPlatform=1.7
+targetPlatform=1.8
```
Comment (lukecwik) [inline maven-archetypes/examples/src/test/resources/projects/basic/archetype.properties]: Can we remove this line in the archetype.properties and instead just use the @java.version@?

Ditto on maven-archetypes/starter/src/main/resources/META-INF/maven/archetype-metadata.xml and maven-archetypes/starter/src/main/resources/archetype-resources/pom.xml

This should allow you to remove all references to targetPlatform everywhere.

Comment (huygaa11) [reply maven-archetypes/examples/src/test/resources/projects/basic/archetype.properties]: When I removed and used @java.version@, the target pom is picking up my local java version, not 1.8

File: pom.xml (hunk 1 above)
Comment (huygaa11) [reply pom.xml]: removed

File: maven-archetypes/examples/src/test/resources/projects/basic/archetype.properties (hunk 2 above)
Comment (huygaa11) [reply maven-archetypes/examples/src/test/resources/projects/basic/archetype.properties]: ready to merge

Comment (lukecwik) [reply maven-archetypes/examples/src/test/resources/projects/basic/archetype.properties]: I see

**Comment by lukecwik** [comment]
The CLA check failed, can you resolve?

**Pull request merged**